cmd/opampsupervisor/e2e_test.go

```diff
@@ -699,6 +716,29 @@ func TestSupervisorOpAMPConnectionSettings(t *testing.T) {
 	}, 10*time.Second, 500*time.Millisecond, "Collector did not connect to new OpAMP server")
 }
 
+func TestSupervisorCanStartWithoutServerThenConnectLater(t *testing.T) {
+	var connectedToServer atomic.Bool
+	initialServer := newUnstartedOpAMPServer(
+		t,
+		defaultConnectingHandler,
+		server.ConnectionCallbacksStruct{
+			OnConnectedFunc: func(_ context.Context, _ types.Connection) {
+				connectedToServer.Store(true)
+			},
+		})
+	defer initialServer.shutdown()
+
+	s := newSupervisor(t, "accepts_conn", map[string]string{"url": initialServer.addr})
+	defer s.Shutdown()
+
+	time.Sleep(11 * time.Second) // We wait until the supervisor gives up on connecting
+	initialServer.start()
+	// TODO FIX
+	result := true
+	waitForSupervisorConnection(initialServer.supervisorConnected, true)
+	require.True(t, result, "Supervisor failed to connect to OpAMP server")
```

Comment:
shouldn't this test use the connectedToServer above?

Reply:
ah yes great observation. I've fixed the test. PTAL.